ghost/core/core/server/services/members/jobs/jobs.js

```diff
@@ -0,0 +1,27 @@
+const path = require('path');
+const jobsService = require('../../jobs');
+
+let hasScheduled = false;
+
+module.exports = {
+    async scheduleExpiredCompCleanupJob() {
+        if (
+            !hasScheduled &&
+            !process.env.NODE_ENV.startsWith('test')
+        ) {
+            // use a random seconds value to avoid spikes to external APIs on the minute
+            const s = Math.floor(Math.random() * 60); // 0-59
+
+            // Run everyday at 12:05:X AM to clean all expired complimentary subscriptions
+            jobsService.addJob({
+                at: `${s} 5 0 * * *`,
+                job: path.resolve(__dirname, 'clean-expired-comped.js'),
+                name: 'clean-expired-comped'
+            });
+
+            hasScheduled = true;
```

Comment:
The method only ever runs once in the Ghost process lifecycle - during the boot during `members.init()` execution. Why do we need this protective code from running the method multiple times? If it is run multiple times does it signal something wrong is going on?

Reply:
🤔 So i picked that pattern from email analytics, which I get has this check because it is run either on init or sometime down the line based on if emails need to be fetched. I am hoping comp check will also move in a similar direction maybe in the cleanup, so i kept the pattern intact as there is no reason to not have it imo. 
I don't think it will be run multiple times right now so i am ok either ways on keeping or removing this check, but also i don't think there is any harm in keeping it as well :)